### ISSUE TITLE
Flux notification controller with github enterprise

### ISSUE THREAD

**Issue by sairamvoggu1007** [fluxcd/notification-controller#199]
Hi Team and @phillebaba 
this is the continuation of previous request: https://github.com/fluxcd/notification-controller/issues/161
Notification controller image updated to :
> image: ghcr.io/fluxcd/notification-controller:v0.13.0

When I'm trying to create github alert observing following issue in notification controller logs 

> kind":"Kustomization","name":"appname","namespace":"platform"}
> {"level":"error","ts":"2021-05-13T08:35:41.567Z","logger":"event-server","msg":"failed to send notification","reconciler kind":"Kustomization","name":"appname","namespace":"platform","error":"Revision string format incorrect: environment/dv01aws/e0358f0de99e1a078105b05ec2e18a8f242aef5694a"}

the github authentication is through SSO that's why I set up the gitrepsotiory resource with SSH (URL and creds) however the  Provider resource has HTTPS address

**Will it be a reason for this kind of issue?**

 Also I didn't find in the doc https://fluxcd.io/docs/guides/notifications/ that github token should be base64 encoded or not?

**Comment by stefanprodan** [comment]
I don't see how is this related to GitHub auth. The revision parser has a bug, it fails to detect that the branch is `environment/dv01aws` and the commit is `e0358f0de99e1a078105b05ec2e18a8f242aef5694a `.

**Comment by sairamvoggu1007** [comment]
@stefanprodan so I updated branch from environment/dv01aws to master for testing thinking the issue could be the '/' within in the branch name.

After updating branch to master seeing different error

> 
> > {"level":"error","ts":"2021-05-13T14:43:17.484Z","logger":"event-server","msg":"failed to send notification","reconciler kind":"Kustomization","name":"appname","namespace":"platform","error":"could not list commit statuses: Get \"https://github.mycompany.com/api/v3/repos/s-voggu/myrepname/commits/b2acde7 │
> > │ c62007333a53c0a8c4f6c05f69907525d/statuses?per_page=50\": net/http: invalid header field value \"Bearer u\\xcd\\xf5\\xd5\\xed\\xb8i\\xceuoV\\xf5\\u007fm\\xba\ │
> > │ \u007fM\\xddkO{\\xd5\\xd7{q\\xbd\\x9d\\xf5\\xbd\\xf5\" for key Authorization"}

**Comment by stefanprodan** [comment]
Ok so these are 2 different issues, the first one is a revision parse bug, the 2nd one looks like the token is encoded twice.

**Comment by sairamvoggu1007** [comment]
@stefanprodan for second issue does  github token should be base64 encoded or not?

**Comment by stefanprodan** [comment]
No, Kubernetes does the encoding for you, run this:
```
kubectl create secret generic gh-token \
--from-literal=token=GH-TOKEN-AS-PASTED-FROM-UI
```

Make sure there is no `\n\r` at the end of the token.

**Comment by stefanprodan** [comment]
You need to set the `certSecretRef `, see https://fluxcd.io/docs/components/notification/provider/#self-signed-certificates

**Comment by sairamvoggu1007** [comment]
@stefanprodan I created secret name tls-certs with ca.crt file and I'm referring token and .crt as following in provider resource. please correct if I'm wrong 

> apiVersion: notification.toolkit.fluxcd.io/v1beta1
> kind: Provider
> metadata:
>   name: github-environment-notification
> spec:
>   type: github
>   address: https://github.mycompany.com/s-voggu/reponame
>   secretRef:
>     # token
>     name: githubsai
>     # tls-cert
>     name: tls-certs

**Comment by sairamvoggu1007** [comment]
Thanks @stefanprodan  above **cerSecretRef** worked and able to get alerts if branch name doesn't have any '/' 

In our case we have branches like "enviornment/dev" environment/test" environmet/prod" with these branch names in notification controller we are observing **Revision string format incorrect:** above you mentioned it as a bug. Could you please let us know once this bug is fixed we can test it

**Comment by phillebaba** [comment]
I will have a look at this and fix the revision parser so that it can deal with more formats.

**Comment by sairamvoggu1007** [comment]
thanks @phillebaba any ETA?

**Comment by sairamvoggu1007** [comment]
@phillebaba @stefanprodan any ETA on this issue when this fix can be released?

**Comment by hiddeco** [comment]
As much as this may be a high priority issue to you personally @sairamvoggu1007; please refrain from pinging maintainers for ETAs.

Given the people responsible to get the pull request merged are already watching the repository activity and receive notifications for review requests, this only causes additional noise and will generally not result in acceleration of closing the issue, especially not if done within a 48h window.

Thank you :sunflower:

**Comment by sairamvoggu1007** [comment]
may I know to what version I need to update notification-controller docker image to get these changes work

**Comment by hiddeco** [comment]
There is no release yet that includes these changes.

**Comment by sairamvoggu1007** [comment]
may I know expected release date with these changes?

**Comment by hiddeco** [comment]
This has been released in `v0.14.1`.

**Comment by sairamvoggu1007** [comment]
thanks very much!. I will be updating notification controller image to v0.1.4.1
